src/holofuel/components/layout/PrimaryLayout/PrimaryLayout.js

```diff
@@ -33,6 +36,21 @@ function PrimaryLayout ({
   const hamburgerClick = () => setMenuOpen(!isMenuOpen)
   const handleMenuClose = () => setMenuOpen(false)
 
+  const [hasUpdatedCounterpartyList, setHasUpdatedCounterpartyList] = useState(false)
+  const { counterpartyList, setCounterpartyList } = useCounterpartyListContext()
+
+  useEffect(() => {
+    // eslint-disable-next-line
+    if (hasUpdatedCounterpartyList) return
+    else if (!isEmpty(actionableTransactions)) {
+      findnewCounterpartiesFromList(actionableTransactions, counterpartyList)
+        .then(newCounterparties => {
+          setCounterpartyList([...counterpartyList, ...newCounterparties])
+          setHasUpdatedCounterpartyList(true)
```

Comment:
I'm not sure if we want to use context for this. I was hoping we could use the Apollo cache for this. Otherwise we're starting to roll our own cache out of contexts, which we want to avoid. Let's talk about the possibility of using Apollo cache